rpcs3/util/serialization.hpp

```diff
@@ -108,12 +108,12 @@ namespace utils
 		{
 			if (m_is_writing) return;
 
-			const u32 offset1 = ::offset32(first) + sizeof(first);
+			const u32 offset1 = ::offset32(first) + sizeof(T);
 			const u32 offset2 = ::offset32(second);
 
-			AUDIT(offset2 >= offset1);
+			AUDIT(::offset32(first) <= ::offset32(second));
 
-			if (offset2 > offset1)
+			if (::offset32(first) <= ::offset32(second))
```

Comment:
Why this? this makes no sense

Reply:
This is because of c++20's [[no_unique_address]].

Reply:
It doesn't make sense still.
And your explanation doesn't tell me anything

Reply:
This attribute allows two distinguished members to have the same offset,

Reply:
It still makes no sense to change this condition if you look at the code in the scope

Reply:
I see, I only need to change the assert.